dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java

```diff
@@ -473,7 +473,7 @@ public CloseableClient build() {
             SSLSocketFactory rawSocketFactory = conf.sslSocketFactory();
             SSLConnectionSocketFactory sslSocketFactory = new SSLConnectionSocketFactory(
                     MetricRegistries.instrument(conf.taggedMetricRegistry(), rawSocketFactory, name),
-                    new String[] {"TLSv1.2"},
+                    TlsProtocols.enabledFor(name),
```

Comment:
So this is dialogue clients announcing TLS1.3 support to servers, 10% of the time. Then for the lifetime of the dialogue client that decision will stick for all connections it creates. On restart there's a chance that we won't announce 1.3 support, but it's also likely it will again be chosen for 1.3.

@carterkozak hard for me to say whether that's better than deterministically switching (based on client hash); I think it's nicer to always fail, as opposed to sometimes fail, but not strong preference.

Reply:
I went back and forth on this quite a few times before I came to the current approach, though I'm still skeptical of the partial rollout strategy here :-/

The reason I switched away from deterministic selection based on the name was that if a given name hashes to a value we select, _all_ dialogue clients will begin to use TLSv1.3 after they upgrade. This means we would lose the slow-ramp in favor of opting in client+server pairs together, which isn't dramatically different from enabling TLSv1.3 across the board.